pkg/operator/events/recorder.go

```diff
@@ -54,7 +54,11 @@ func GetControllerReferenceForCurrentPod(client kubernetes.Interface, targetName
 		// If that fails, lets try to guess the pod by listing all pods in namespaces and using the first pod in the list
 		reference, err := guessControllerReferenceForNamespace(client.CoreV1().Pods(targetNamespace))
 		if err != nil {
-			return nil, err
+			// If this fails, do not give up with error but instead use the namespace as controller reference for the pod
+			// NOTE: This is last resort, if we see this often it might indicate something is wrong in the cluster.
+			//       In some cases this might help with flakes.
+			klog.Warningf("Unable to guess owner reference in namespace %q (will use namespace instead): %v", targetNamespace, err)
+			return getControllerReferenceForNamespace(targetNamespace)
 		}
 		return GetControllerReferenceForCurrentPod(client, targetNamespace, reference)
```

Comment:
change signature of this method to never fail or return the error and a valid reference.

Reply:
I hate having to set /var/run/.../namespace to test locally.  It would be better if that is absent to report to openshift-config or something.

Reply:
@deads2k ok, I changed the signature to return error but also fallback reference to namespace

@smarterclayton you don't have to create any file, the namespace is provided as argument to the function, so it can be `foo`.